File: app/src/main/java/com/stedi/randomimagegenerator/app/view/fragments/base/StepFragment.java
package com.stedi.randomimagegenerator.app.view.fragments.base;

import android.support.annotation.CallSuper;
import android.support.annotation.NonNull;

import com.stedi.randomimagegenerator.app.other.CommonKt;
import com.stepstone.stepper.BlockingStep;
import com.stepstone.stepper.StepperLayout;
import com.stepstone.stepper.VerificationError;

public abstract class StepFragment extends ButterKnifeFragment implements BlockingStep {
    @Override
    public VerificationError verifyStep() {
        return null;
    }

    @Override
    public void onSelected() {
    }

    @Override
    public void onError(@NonNull VerificationError error) {
    }

    @CallSuper
    @Override
    public void onNextClicked(StepperLayout.OnNextClickedCallback callback) {
        CommonKt.hideInput(getActivity());
        callback.goToNextStep();
    }

    @CallSuper
    @Override
    public void onCompleteClicked(StepperLayout.OnCompleteClickedCallback callback) {
        callback.complete();
    }

    @CallSuper
    @Override
    public void onBackClicked(StepperLayout.OnBackClickedCallback callback) {
        CommonKt.hideInput(getActivity());
        callback.goToPrevStep();
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/view/fragments/base/StepFragment.kt
package com.stedi.randomimagegenerator.app.view.fragments.base

import android.support.annotation.CallSuper
import com.stedi.randomimagegenerator.app.other.hideInput
import com.stepstone.stepper.BlockingStep
import com.stepstone.stepper.StepperLayout
import com.stepstone.stepper.VerificationError

abstract class StepFragment : ButterKnifeFragment(), BlockingStep {

    override fun verifyStep(): VerificationError? = null

    override fun onSelected() {}

    override fun onError(error: VerificationError) {}

    @CallSuper
    override fun onNextClicked(callback: StepperLayout.OnNextClickedCallback) {
        activity?.hideInput()
        callback.goToNextStep()
    }

    @CallSuper
    override fun onCompleteClicked(callback: StepperLayout.OnCompleteClickedCallback) {
        callback.complete()
    }

    @CallSuper
    override fun onBackClicked(callback: StepperLayout.OnBackClickedCallback) {
        activity?.hideInput()
        callback.goToPrevStep()
    }
}